### ISSUE TITLE
Dropdown: Accessibility issue - Lost focus on filter search using keyboard

### ISSUE THREAD

**Issue by donSan4o** [primefaces/primereact#7202]
### Describe the bug

focusOnItem function, that was added in this PR:
https://github.com/primefaces/primereact/pull/7175
broke keyboard control, lost focus on search input

### System Information

```Shell
primereact 10.8.3
```

### Steps to reproduce the behavior

1. Go to https://primereact.org/dropdown/#filter
2. Open dropdown with keyboard
3. Try to type in search input

### Expected behavior

Focus should be always on on search input
